core/src/commonMain/kotlin/com/sourcepoint/mobile_core/Coordinator.kt

```diff
@@ -38,4 +61,255 @@ class Coordinator(
+                    messageId = action.messageId,
+                    consentAllRef = postPayloadFromGetCall?.consentAllRef,
+                    vendorListId = postPayloadFromGetCall?.vendorListId,
+                    pubData = action.encodablePubData,
+                    pmSaveAndExitVariables = action.pmPayload,
+                    sendPVData = state.gdprMetaData?.wasSampled ?: false,
+                    propertyId = propertyId,
+                    sampleRate = state.gdprMetaData?.sampleRate,
+                    idfaStatus = idfaStatus,
+                    granularStatus = postPayloadFromGetCall?.granularStatus,
+                    includeData = includeData
+                )
+            )
+        }
+        catch (error: Throwable) {
+            throw error
+        }
+
+    override suspend fun postChoiceCCPA(action: SPAction): CCPAChoiceResponse =
+        try {
+            spClient.postChoiceCCPAAction(
+                actionType = action.type,
+                request = CCPAChoiceRequest(
+                    authId = authId,
+                    uuid = state.ccpa?.uuid,
+                    messageId = action.messageId,
+                    pubData = action.encodablePubData,
+                    pmSaveAndExitVariables = action.pmPayload,
+                    sendPVData = state.ccpaMetaData?.wasSampled?: false,
+                    propertyId = propertyId,
+                    sampleRate = state.ccpaMetaData?.sampleRate,
+                    includeData = includeData
+                )
+            )
+        }
+        catch (error: Throwable) {
+            throw error
+        }
+
+    override suspend fun postChoiceUSNat(action: SPAction): USNatChoiceResponse =
+        try {
+            spClient.postChoiceUSNatAction(
+                actionType = action.type,
+                request = USNatChoiceRequest(
+                    authId = authId,
+                    uuid = state.usNat?.uuid,
+                    messageId = action.messageId,
+                    vendorListId = state.usNatMetaData?.vendorListId,
+                    pubData = action.encodablePubData,
+                    pmSaveAndExitVariables = action.pmPayload,
+                    sendPVData = state.usNatMetaData?.wasSampled?: false,
+                    propertyId = propertyId,
+                    sampleRate = state.usNatMetaData?.sampleRate,
+                    idfaStatus = idfaStatus,
+                    granularStatus = state.usNat?.consentStatus?.granularStatus,
+                    includeData = includeData
+                )
+            )
+        }
+        catch (error: Throwable) {
+            throw error
+        }
+
+    private fun handleGPDRPostChoice(
+        action: SPAction,
+        getResponse: ChoiceAllResponse?,
+        postResponse: GDPRChoiceResponse
+    ) {
+        if (action.type == SPActionType.SaveAndExit) {
+            state.gdpr?.tcData = postResponse.tcData?: emptyMap()
+        }
```

Comment:
@Nevazhnovu let's move this `if` statement _after_ the `state.gdpr` assignment below, because this way we're sure the `state.gdpr` is not null.